scripts/histmakers/mw_with_mu_eta_pt.py

```diff
@@ -783,19 +800,29 @@ def build_graph(df, dataset):
 
     return results, weightsum
 
-resultdict = narf.build_and_run(datasets, build_graph)
-if not args.onlyMainHistograms and args.muonScaleVariation == 'smearingWeightsGaus' and not isFloatingPOIsTheoryAgnostic:
-    logger.debug("Apply smearingWeights")
-    muon_calibration.transport_smearing_weights_to_reco(
-        resultdict,
-        smearing_weights_procs,
-        nonClosureScheme = args.nonClosureScheme
-    )
-if args.validationHists:
-    muon_validation.muon_scale_variation_from_manual_shift(resultdict)
-
-if not args.noScaleToData:
-    scale_to_data(resultdict)
-    aggregate_groups(datasets, resultdict, groups_to_aggregate)
-
-output_tools.write_analysis_output(resultdict, f"{os.path.basename(__file__).replace('py', 'hdf5')}", args)
+if args.sequentialEventLoops:
+    dataset_sets = [[dataset] for dataset in datasets]
+else:
+    dataset_sets = [datasets]
+
+for loop_datasets in dataset_sets:
+    resultdict = narf.build_and_run(loop_datasets, build_graph)
+    if not args.onlyMainHistograms and args.muonScaleVariation == 'smearingWeightsGaus' and not isFloatingPOIsTheoryAgnostic:
+        logger.debug("Apply smearingWeights")
+        muon_calibration.transport_smearing_weights_to_reco(
+            resultdict,
+            smearing_weights_procs,
+            nonClosureScheme = args.nonClosureScheme
+        )
+    if args.validationHists:
+        muon_validation.muon_scale_variation_from_manual_shift(resultdict)
+
+    if not args.noScaleToData and not args.sequentialEventLoops:
+        scale_to_data(resultdict)
```

Comment:
Don't we still have to apply the scaling to data and the sum of subsamples (e.g. ttbar, single top, ... into Top) at some point outside the loop ?

Reply:
The scaling happens automatically later in setupCombine if it's not done here.

It can't be easily done here because the data is not present in the result dictionary so the lumi is not accessible.

Reply:
ok for the data scaling, but what about the group aggregation ? Being a post processing action it might still be useful to do it before running setupCombine, otherwise the making of fakes might be slower since all subsamples and their histograms have to be read and subtracted from data rather than directly their sum, or at least I think this was one of the motivations to implement it

Reply:
The aggregation is not possible for the same reason.  Since the whole point is that you never have the histograms from multiple processes in memory.

We can live with the slower handling in setupCombine for this special case I think.
(by default the event loops are still run all together and the scaling and aggregation are still applied of course)

Reply:
In principle we could scale the cross sections, but not the data, in this case we could also aggregate. And I think it should even work out of the box, because whatever scale is applied is propagated into the sum of weights and in the setupCombine all what was not applied, should be applied.

Reply:
The aggregation really can't work because the histograms from the previous processes are not in memory any more.

For the xsec scaling is there an existing function which can be called?

Reply:
Ah right about the aggregation. In this case, I think we don't gain anything from the xsec scaling alone since we would still have to scale the histogram in setupCombine to the lumi (we would only save a scalar multiplication, which is negligible). So I'm fine with merging as it is.